### PR TITLE
fix(ci): don't run publish job unless it's the main branch

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    workflows: ["Continuous integration"]
+  merge_group:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
```
Comment (PhilippeMts) [inline .github/workflows/publish.yml:14]: isn't there some redundancy between the `workflow_run` event and the other ones?

**Review by PhilippeMts — APPROVED** [review]
thanks for this fix.

you'll find a couple of inline comments. otherwise, we are good to go, provided that we follow the _hotfix_ git flow strategy when proceeding to merges related to this PR.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (jxs) [reply .github/workflows/publish.yml:14]: no, `workflow_run` is the `depends` intra workflows, this is stating that this workflow should only run if `Continuous integration` execute and passes. See [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) for more info

Comment (PhilippeMts) [reply .github/workflows/publish.yml:14]: thanks joão. your point is really clear.
sorry mine was probably lacking some clarity, questioning the very need for the `merge_group`, `pull_request` and `push` triggers, next to the `workflow_run` condition.

looking forward to your answer.

anyway, this is not blocking to me 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,15 +1,12 @@
-name: Continuous integration & Continuous deployment
+name: Continuous integration
 
 on:
   merge_group:
-    branches:
-      - main
   pull_request:
-    branches:
-      - main
```
[the remark this answers is not in the document]
Comment (jxs) [reply .github/workflows/ci.yml:9]: sorry Philippe, did not understand. Can you expand?

[the remark this answers is not in the document]
Comment (PhilippeMts) [reply .github/workflows/ci.yml:9]: absolutely : 

- having the filter on the `main` branch for all three types of events makes sense to me.
- having no filter on any branch for all three types of events makes sense to me too.
- but having such a filter only on the `push` event does make less sense I have to say.

am I misreading sthg maybe?

[the remark this answers is not in the document]
Comment (jxs) [reply .github/workflows/ci.yml:9]: Thanks Philippe, so:
- having the filter on the main branch for all three types of events makes sense to me.
this does not make sense imho, cause we want to run the CI workflows on every PR, but I agree that if no filter for `merge_group` we should also not have for `push`. So I'd say let's have `merge_queue` only on main as well, wdy?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (PhilippeMts) [reply .github/workflows/publish.yml:14]: I totally subscribe to your description of our goal 👍🙏

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jxs) [reply .github/workflows/ci.yml]: ok, updated!

**Pull request merged**